File: BadBlock.py
makeneutral = {"fuck":"sticks", "shit":"trash", "bitch":"dingledoo" }
makegood = {"fuck":"love","shit":"sunshine", "bitch":"happy" }
modes = ["MAKENEUTRAL", "MAKEGOOD"]
def mode():
  choice = input("[1]: MakeNeutral or [2]: MakeGood")
  
  print(choice)
  
  current = ""
  
  if choice == 1:
    current = modes[0]
    print(current)
  elif choice == 2:
    current = modes[2]
    print(current)
  
  print('Your current mode is: ' + current)


#   fuckprint("phrase: " + phrase)
  
def phrase_fix(dictionary):
  phrase = input("Insert phrase to BadBlock here: ")
  words = phrase.split()
  for x in range(len(words)):
      badword = words[x]
      if (badword in dictionary):
        words[x] = dictionary.get(badword)
        new_phrase = " ".join(words)
  print(new_phrase)

phrase_fix(makeneutral)

def bad_words_store():
    f = open("badwordstwo.csv", "r")
    mySet = set()
    for x in f:
        mySet.add(x)
    return mySet
  
def phrase_fix(dictionary):
  badSet = bad_words_store()
  phrase = input("Insert here: ")
  words = phrase.split()
  for x in range(len(words)):
      badword = words[x]
      if (badword in badSet):
        words[x] = wordReplace(badword)
        new_phrase = " ".join(words)
  print(new_phrase)

def wordReplace(word):
  switcher = {
    fuck: return "diddle"
    fucker: return "diddler"
    fucking: return "diddling"
    motherfucker: return "mother flubber"
    bitch: return "snitch"
    nigger or nigga: return "neighbour"
    dickpenis: return "male genitalia"
    pussy or vagina or cunt: return "female genitalia"
    else: return "BLEEP"
  }

File: BadBlock.js
// const map = {"bitch": "beep", "fuck": "yo"}
// // fill this map with your python map
// checkWord = function(word){
//    if(map.hasOwnProperty(word)){
//       //Your replace code here
//       console.log(map[word]);
//    }
// }
// checkWord("bitch");
// checkWord("yo");
// checkWord("fuck");

var json = require('badwordsjson.json')

let badwordset;

var toReplace = $('body').text()

badWordSetAdder = function(word){
   for (var key in json) {
      badwordset.add(key);
   }
   return badwordset
}

wordReplace = function(word){
   switch (word){
      default: return "BLEEP"
      case "fuck*": return "diddle"
      case "fucker": return "diddler"
      case "fucking": return "diddling"
      case "motherfucker": return "mother flubber"
      case "bitch": return "brat"
      case ("nigger"|"nigga"): return "neighbour"
      case ("dick*"|"penis"): return "male genitalia"
      case ("pussy"|"vagina"|"cunt"): return "female genitalia"
      }
   }

phrase_fix = function(text){
   let words = text.split();
   let badwordset = badWordSetAdder(json);
   
   for (let x in badwordset) {
     let badword = words[x];
     if (badwordset.contains(badword)){
       words[x] = wordReplace(badword);
       let new_text = words.join();
     }
   }
   return new_text;
 }


 document.getElementById("butt").addEventListener("click", phrase_fix(toReplace));

